src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -971,31 +954,57 @@ private void AddEasyHandle(RequestCompletionSource state)
             }
         }
 
+        private static void RemoveEasyHandle(SafeCurlMultiHandle multiHandle, GCHandle stateHandle, RequestCompletionSource state)
+        {
+            RemoveEasyHandle(multiHandle, state, false);
+            if (stateHandle.IsAllocated)
+            {
+                stateHandle.Free();
+            }
+        }
+
         private static void RemoveEasyHandle(SafeCurlMultiHandle multiHandle, GCHandle stateHandle, bool onMultiStack)
         {
+            Debug.Assert(stateHandle.IsAllocated);
+
             RequestCompletionSource state = (RequestCompletionSource)stateHandle.Target;
+            RemoveEasyHandle(multiHandle, state, onMultiStack);
+            stateHandle.Free();
```

Comment:
Actually, I may have given bad advice here.  Is this easy handle still potentially running and causing events to be invoked?  If so, we don't want to free the state handle until after we're sure no more callbacks could be invokd with the context ptr.